src/core/syscalls/syscall_handler.rs

```diff
@@ -145,11 +144,27 @@ pub(crate) trait SyscallHandler {
             "library_call" => LibraryCallStruct::from_ptr(vm, syscall_ptr),
             "get_caller_address" => GetCallerAddressRequest::from_ptr(vm, syscall_ptr),
             "get_sequencer_address" => GetSequencerAddressRequest::from_ptr(vm, syscall_ptr),
+            "get_block_number" => GetBlockNumberRequest::from_ptr(vm, syscall_ptr),
             "get_block_timestamp" => GetBlockTimestampRequest::from_ptr(vm, syscall_ptr),
             _ => Err(SyscallHandlerError::UnknownSyscall),
         }
     }
 
+    fn get_block_number(
+        &mut self,
+        vm: &mut VirtualMachine,
+        syscall_ptr: Relocatable,
+    ) -> Result<(), SyscallHandlerError> {
+        self._read_and_validate_syscall_request("get_block_number", vm, syscall_ptr)?;
+
+        self._write_syscall_response(
+            &GetBlockNumberResponse::new(self.get_block_info().block_number),
+            vm,
+            syscall_ptr,
+        )?;
+        Ok(())
```

Comment:
as an standard we are calling `write_syscall_response` directly from the response struct, like this 

`let response = GetBlockNumberResponse::new(self.get_block_info().block_number)`

`        response.write_syscall_response(vm, syscall_ptr)`

i think that we maybe we should delete `._write_syscall_response` as it is acting as an unnecessary wrapper.